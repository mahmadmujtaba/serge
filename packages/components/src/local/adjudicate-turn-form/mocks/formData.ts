import { AdjudicateTurnFormPopulate } from "@serge/custom-types"

const formData: AdjudicateTurnFormPopulate = {
  contactId: 'C234',
  status: [
    {
      name: 'Fishing',
      mobile: false
    },
    {
      name: 'Moored',
      mobile: false
    },
    {
      name: 'Transiting',
      mobile: true
    }
  ],
  speed: [10, 20, 30],
  visibleTo: [
    {
      name: 'Blue Force',
      colour: '#69c'
    },
    {
      name: 'Red Force',
      colour: '#f00'
    },
    {
      name: 'White Force',
      colour: '#fff'
    }
  ],
  condition: ['Working', 'Disabled', 'Immobile', 'Destroyed']
}

export default formData
