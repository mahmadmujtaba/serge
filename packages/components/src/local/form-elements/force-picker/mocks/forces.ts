const forces = [
  {
    name: 'Blue Force',
    colour: '#69c',
    selected: true
  },
  {
    name: 'Red Force',
    colour: '#f00',
    selected: false
  },
  {
    name: 'White Force',
    colour: '#fff',
    selected: false
  }
]

export default forces
