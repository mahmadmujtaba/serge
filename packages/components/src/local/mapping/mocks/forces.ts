export const forces = [
    {
      color: '#FCFBEE',
      dirty: false,
      icon: 'default_img/umpireDefault.png',
      name: 'White',
      overview: 'Umpire force.',
      roles: [
        {
          control: true,
          isInsightViewer: true,
          isObserver: true,
          name: 'Game Control',
          password: 'p2311'
        }
      ],
      umpire: true,
      uniqid: 'umpire'
    },
    {
      assets: [
        {
          name: 'HMS DEVONSHIRE',
          contactId: 'C043',
          condition: 'Full capability',
          perceptions: {},
          platformType: 'frigate',
          position: 'S23',
          status: {
            state: 'Transiting',
            speedKts: 20
          },
          uniqid: 'a0pra00001'
        },
        {
          name: 'STING',
          contactId: 'C072',
          condition: 'Full capability',
          perceptions: {},
          platformType: 'helicopter',
          position: 'P01',
          status: {
            state: 'Landed'
          },
          uniqid: 'a0pra00002'
        },
        {
          name: 'P8',
          contactId: 'C076',
          condition: 'Full capability',
          perceptions: {},
          platformType: 'fixed-wing-aircraft',
          position: 'P03',
          status: {
            state: 'Landed'
          },
          uniqid: 'a0pra00012'
        },
        {
          name: 'EDDYSTONE PT',
          contactId: 'C012',
          condition: 'Full capability',
          perceptions: {
            Red: {
              force: 'Blue',
              type: ''
            }
          },
          platformType: 'merchant-vessel',
          position: 'S23',
          status: {
            state: 'Transiting',
            speedKts: 20
          },
          uniqid: 'a0pra00003'
        }
      ],
      color: '#00F',
      dirty: false,
      icon: 'default_img/umpireDefault.png',
      name: 'Blue',
      overview: 'Blue force.',
      roles: [
        {
          control: false,
          isInsightViewer: false,
          isObserver: false,
          name: 'CO',
          password: 'p5543'
        }
      ],
      umpire: false,
      uniqid: 'Blue'
    },
    {
      assets: [
        {
          name: 'TARRADA',
          contactId: 'C065',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {},
          platformType: 'fishing-vessel',
          position: 'N07',
          status: {
            state: 'Transiting',
            speedKts: 10
          },
          uniqid: 'a0pra000100'
        },
        {
          name: 'WBIED-1',
          contactId: 'C108',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {},
          platformType: 'torpedo',
          position: 'T01',
          status: {
            state: 'Onboard'
          },
          uniqid: 'a0pra020102'
        },
        {
          name: 'WBIED-2',
          contactId: 'C118',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {},
          platformType: 'torpedo',
          position: 'T03',
          status: {
            state: 'Onboard'
          },
          uniqid: 'a0pra130102'
        },
        {
          name: 'Osa',
          contactId: 'C003',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {
            Blue: {
              force: 'Red',
              type: ''
            }
          },
          platformType: 'fast-attack-craft',
          position: 'M05',
          status: {
            state: 'Moored'
          },
          uniqid: 'a0pra160102'
        },
        {
          name: 'Yevgenya',
          contactId: 'C313',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {
            Blue: {
              force: 'Red',
              type: 'fast-attack-craft'
            }
          },
          platformType: 'mcmv',
          position: 'M04',
          status: {
            state: 'Moored'
          },
          uniqid: 'a0pra170102'
        },
        {
          name: 'Coastal Radar C',
          contactId: 'C476',
          condition: 'Full capability',
          locationPending: true,
          perceptions: {},
          platformType: 'coastal-radar-site',
          position: 'Q12',
          status: {
            state: 'Inactive'
          },
          uniqid: 'a0prc000103'
        }
      ],
      color: '#F00',
      dirty: false,
      icon: 'default_img/umpireDefault.png',
      name: 'Red',
      overview: 'Red force.',
      roles: [
        {
          control: false,
          isInsightViewer: false,
          isObserver: false,
          name: 'CO',
          password: 'p3244'
        }
      ],
      umpire: false,
      uniqid: 'Red'
    }
  ]


export default forces
