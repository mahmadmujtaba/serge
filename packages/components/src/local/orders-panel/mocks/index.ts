import { ADJUDICATION_PHASE } from '@serge/config'

const data = {
  selectedForce: 'blue',
  phase: ADJUDICATION_PHASE
}

export default data
