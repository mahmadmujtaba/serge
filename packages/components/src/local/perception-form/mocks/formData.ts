import { PerceptionFormData } from '@serge/custom-types'

const formData: PerceptionFormData = {
  populate: {
    perceivedForces: [
      {
        name: 'Blue Force',
        colour: '#69c'
      },
      {
        name: 'Red Force',
        colour: '#f00'
      },
      {
        name: 'White Force',
        colour: '#fff'
      }
    ],
    perceivedTypes: [
      { uniqid: 'aa', name:'Fishing vessel'},
      { uniqid: 'a1', name:'torpedo'},
      { uniqid: 'a2', name:'Boghammer'},
      { uniqid: 'a3', name:'Frigate'}
    ]
  },
  values: {
    perceivedNameVal: 'Test',
    perceivedForceVal: 'blue-force',
    perceivedTypeId: 'a3',
    assetId: 'c00012',
    iconURL: 'frigate.svg'
  }
}

export default formData
