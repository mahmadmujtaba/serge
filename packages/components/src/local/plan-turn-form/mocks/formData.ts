import { COMMODITY_TYPE_NUMBER, COMMODITY_VALUE_NUMBER } from '@serge/config'
import { PlanTurnFormData } from '@serge/custom-types'

const formData: PlanTurnFormData = {
  populate: {
    status: [{
      name: 'Moored',
      mobile: false
    },
    {
      name: 'Transiting',
      mobile: true
    }],
    speed: [10, 20, 30],
    attributes: [{ commId: 'comm_a', commType: COMMODITY_TYPE_NUMBER, name: 'Fuel', editableByPlayer: false },
      { commId: 'comm_b', commType: COMMODITY_TYPE_NUMBER, name: 'People', editableByPlayer: false },
      { commId: 'comm_c', commType: COMMODITY_TYPE_NUMBER, name: 'Water', units: 'litres', editableByPlayer: false }]
  },
  values: {
    statusVal: {
      name: 'Transiting',
      mobile: true
    },
    speedVal: 10,
    turnsVal: 5,
    condition: 'Working',
    attributes: [{ commId: 'comm_a', commType: COMMODITY_VALUE_NUMBER, value: 12 },
      { commId: 'comm_b', commType: COMMODITY_VALUE_NUMBER, value: 213 },
      { commId: 'comm_c', commType: COMMODITY_VALUE_NUMBER, value: 12450 }]
  }
}

export default formData
