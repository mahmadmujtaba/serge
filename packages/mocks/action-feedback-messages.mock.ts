import { SET_FEEDBACK_MESSAGES } from '@serge/config'
import { SetWargameFeedbackAction } from '@serge/custom-types'
import { feedbackMessage1, feedbackMessage2 } from './feedback-message.mock'

const actionFeedbackMessages: SetWargameFeedbackAction = {
  type: SET_FEEDBACK_MESSAGES,
  payload: [feedbackMessage1, feedbackMessage2]
}

export default actionFeedbackMessages
