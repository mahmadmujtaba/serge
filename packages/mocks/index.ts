/***
* This file is autogenerated, please do not edit it.
* To generate an updated version, please run `yarn genreg`
***/

export { default as MessagesMock } from './messages.mock'
export { default as WargameMock } from './wargame.mock'