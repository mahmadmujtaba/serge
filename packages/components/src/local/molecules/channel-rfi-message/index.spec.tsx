/* global it expect */
import React from 'react'
import renderer from 'react-test-renderer'
import ChannelMessage from './index'

import { GameMessagesMockRFI } from '@serge/mocks'
const [defMessage] = GameMessagesMockRFI

describe('ChannelMessage component:', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ChannelMessage
          role='CO'
          isUmpire={true}
          message={defMessage}
          onRead={_readMessage => {}}
          onChange={_nextMessage => {}}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
