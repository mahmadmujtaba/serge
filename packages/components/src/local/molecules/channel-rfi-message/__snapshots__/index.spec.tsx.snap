// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelMessage component: renders correctly 1`] = `
<div
  className="message-list-wrapper"
  onClick={[Function]}
  style={
    Object {
      "borderColor": "#f00",
    }
  }
>
  <div
    className="main"
    onClick={[Function]}
  >
    <div
      className="main"
    >
      <div
        className="control control-channel"
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          style={
            Object {
              "color": "rgb(24, 255, 255)",
              "fontSize": 20,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </div>
      <div
        className="content"
        onClick={[Function]}
      >
        <div
          className="header"
        >
          <div
            className="row"
          >
            <h2>
              <div
                className="MuiChip-root makeStyles-root-1 makeStyles-root-4"
                onKeyDown={[Function]}
                onKeyUp={[Function]}
              >
                <span
                  className="MuiChip-label makeStyles-label-3 makeStyles-label-5"
                />
              </div>
               
            </h2>
            <div
              className="rfi-info"
            >
              <div
                className="status"
              >
                <p>
                  PENDING
                </p>
              </div>
            </div>
          </div>
          <div
            className="row"
          >
            <div>
              <span
                className="timestamp"
              >
                12:52
              </span>
              <div
                className="MuiChip-root makeStyles-root-1 makeStyles-root-6 MuiChip-sizeSmall makeStyles-sizeSmall-2"
                onKeyDown={[Function]}
                onKeyUp={[Function]}
              >
                <span
                  className="MuiChip-label makeStyles-label-3 makeStyles-label-7 MuiChip-labelSmall"
                >
                  CO
                </span>
              </div>
              <div
                className="MuiChip-root makeStyles-root-1 makeStyles-root-8 MuiChip-sizeSmall makeStyles-sizeSmall-2"
                onKeyDown={[Function]}
                onKeyUp={[Function]}
              >
                <span
                  className="MuiChip-label makeStyles-label-3 makeStyles-label-9 MuiChip-labelSmall"
                >
                  Chat
                </span>
              </div>
              <div
                className="MuiChip-root makeStyles-root-1 makeStyles-root-10 MuiChip-sizeSmall makeStyles-sizeSmall-2"
                onKeyDown={[Function]}
                onKeyUp={[Function]}
              >
                <span
                  className="MuiChip-label makeStyles-label-3 makeStyles-label-11 MuiChip-labelSmall"
                >
                  Unread
                </span>
              </div>
            </div>
            <div
              className="badges"
            >
              
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="main use-indent"
      onClick={[Function]}
      style={Object {}}
    >
      <div
        className="main"
      >
        <div
          className="main theme-dark disabled"
        >
          <label
            htmlFor="question_2020-03-25T15:08:47.525Z"
          >
            Request
          </label>
          <textarea
            disabled={true}
            id="question_2020-03-25T15:08:47.525Z"
            onChange={[Function]}
            value="[message empty]"
          />
        </div>
        <div
          className="main theme-dark disabled"
        >
          <label
            htmlFor="answer_2020-03-25T15:08:47.525Z"
          >
            Answer
          </label>
          <textarea
            disabled={true}
            id="answer_2020-03-25T15:08:47.525Z"
            onChange={[Function]}
            value=""
          />
        </div>
        <div
          className="main theme-dark disabled"
        >
          <label
            htmlFor="private_message_2020-03-25T15:08:47.525Z"
          >
            <svg
              className="svg-inline--fa fa-user-secret"
            />
             
            Private Message
          </label>
          <textarea
            disabled={true}
            id="private_message_2020-03-25T15:08:47.525Z"
            onChange={[Function]}
            value=""
          />
        </div>
        <div
          className="actions"
        >
          <button
            className="MuiButtonBase-root MuiButton-root makeStyles-root-12 makeStyles-root-14 MuiButton-contained makeStyles-contained-13 makeStyles-contained-15 MuiButton-containedSizeSmall MuiButton-sizeSmall"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Take Ownership
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
